rcl_action/src/rcl_action/action_client.c

```diff
@@ -0,0 +1,723 @@
+rcl_action_client_init(
+  rcl_action_client_t * action_client,
+  rcl_node_t * node,
+  const rosidl_action_type_support_t * type_support,
+  const char * action_name,
+  const rcl_action_client_options_t * options)
+{
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Initializing action client");
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  rcl_allocator_t * allocator = (rcl_allocator_t *)&options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(
+    allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(action_client, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(node, RCL_RET_INVALID_ARGUMENT);
+  if (!rcl_node_is_valid(node)) {
+    return RCL_RET_NODE_INVALID;
+  }
+  RCL_CHECK_ARGUMENT_FOR_NULL(type_support, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(action_name, RCL_RET_INVALID_ARGUMENT);
+  RCUTILS_LOG_DEBUG_NAMED(
+    ROS_PACKAGE_NAME, "Initializing client for action name '%s'", action_name);
+  // Allocate space for the implementation struct.
+  rcl_action_client_impl_t *impl = allocator->allocate(
+    sizeof(rcl_action_client_impl_t), allocator->state);
+  RCL_CHECK_FOR_NULL_WITH_MSG(impl, "allocating memory failed", return RCL_RET_BAD_ALLOC);
+
+  impl->action_name = rcutils_strdup(action_name, *((rcutils_allocator_t *)allocator));
+  if (NULL == impl->action_name) {
+    allocator->deallocate(impl, allocator->state);
+    return RCL_RET_BAD_ALLOC;
+  }
+  RCUTILS_LOG_DEBUG_NAMED(
+    ROS_PACKAGE_NAME, "Initializing action goal client");
+  rcl_client_options_t goal_client_options = {
+    .qos = options->goal_service_qos, .allocator = *allocator
+  };
+  rcl_ret_t ret, critical_ret;
+  ret = rcl_action_goal_service_client_init(
+    &impl->goal_client, node, 
+    &type_support->goal_service_type_support,
+    impl->action_name, &goal_client_options);
+  if (RCL_RET_OK == ret) {
+    RCUTILS_LOG_DEBUG_NAMED(
+      ROS_PACKAGE_NAME, "Initializing action cancel client");
+    rcl_client_options_t cancel_client_options = {
+      .qos = options->cancel_service_qos, .allocator = *allocator
+    };
+    ret = rcl_action_cancel_service_client_init(
+      &impl->cancel_client, node,
+      &type_support->cancel_service_type_support,
+      impl->action_name, &cancel_client_options);
+    if (RCL_RET_OK == ret) {
+      RCUTILS_LOG_DEBUG_NAMED(
+        ROS_PACKAGE_NAME, "Action cancel client initialized");
+      rcl_client_options_t result_client_options = {
+        .qos = options->result_service_qos, .allocator = *allocator
+      };
+      RCUTILS_LOG_DEBUG_NAMED(
+        ROS_PACKAGE_NAME, "Initializing action result client");
+      ret = rcl_action_result_client_init(
+        &impl->result_client, node,
+        &type_support->result_service_type_support,
+        impl->action_name, &result_client_options);
+      if (RCL_RET_OK == ret) {
+        RCUTILS_LOG_DEBUG_NAMED(
+          ROS_PACKAGE_NAME, "Action cancel result initialized");
```

Comment:
`cancel result` -> `get result`

Reply:
Fixed in c5dabd05bc66f9c01431ec5cfe1b86815081c823.